Lib/pathlib.py

```diff
@@ -61,6 +61,15 @@ def _is_case_sensitive(flavour):
 # Globbing helpers
 #
 
+
+_FNMATCH_PREFIX, _FNMATCH_SUFFIX = fnmatch.translate('_').split('_')
+_FNMATCH_SLICE = slice(len(_FNMATCH_PREFIX), -len(_FNMATCH_SUFFIX))
```

Comment:
This is a bit of a hack. Alternatively I could add an argument to `fnmatch.translate()` that makes it exclude the prefix/suffix, or add a new private `fnmatch._translate()` function perhaps? Not sure.

Reply:
@AlexWaygood any recommendations? Thanks

Reply:
I wonder if this hack might be more performant than a pass-through function written in Python:

```py
_strip_fnmatch_prefix_and_suffix = operator.itemgetter(slice(_FNMATCH_PREFIX, _FNMATCH_SUFFIX))
```

The `operator` functions are written in C, so that might be pretty fast.

And then lower down, where you currently do `fnmatch.translate(part)[_FNMATCH_SLICE]`, you could do `_strip_fnmatch_prefix_and_suffix(fnmatch.translate(part))` instead.

You'd have to import `operator` at the top of `pathlib`, but I think it may be imported at startup anyway, so it probably wouldn't slow down the import of `pathlib` too much.

Anyway, always worth doing some quick benchmarking! Adding a pass-through Python function here might not actually slow down the code too much; and using `operator` might not speed things up here.

Reply:
Interestingly, using `slice()` without `itemgetter()` seems most performant:

```
$ ./python -m timeit -s 'from fnmatch import translate; pat = translate("*.py"); prefix, suffix = translate("_").split("_")' 'pat[len(prefix):-len(suffix)]'
5000000 loops, best of 5: 86.3 nsec per loop
$ ./python -m timeit -s 'from fnmatch import translate; pat = translate("*.py"); prefix, suffix = translate("_").split("_"); sl = slice(len(prefix), -len(suffix));' 'pat[sl]'
5000000 loops, best of 5: 54.1 nsec per loop
$ ./python -m timeit -s 'from fnmatch import translate; pat = translate("*.py"); from operator import itemgetter; prefix, suffix = translate("_").split("_"); sl = itemgetter(slice(len(prefix), -len(suffix)))' 'sl(pat)'
5000000 loops, best of 5: 62.6 nsec per loop
```

Reply:
I've added a comment giving an explanation for `_FNMATCH_SLICE` - do you think it's sufficient? I think the use of the `slice` object hints towards what's going on too.

Reply:
> I've added a comment giving an explanation for `_FNMATCH_SLICE` - do you think it's sufficient? I think the use of the `slice` object hints towards what's going on too.

Yes, I think that works fine! Won't formally approve as I haven't looked over the PR comprehensively (and don't have the time right now, I have language-summit blogs to finish this evening!!). But the comment looks great, and is very helpful :)

Reply:
Thank you!